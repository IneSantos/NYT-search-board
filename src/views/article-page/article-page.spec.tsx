import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import ArticlePage from "./article-page";

let store;
let mockedStore;

describe("ArticlePage atom tests", () => {
  test(`ArticlePage component render`, () => {
    mockedStore = configureStore();
    store = mockedStore({
      article: {
        article: {
          id: "",
          headline: "This is a headline",
          abstract: "",
          keywords: [],
          section: "",
          snippet: "",
          pub_date: "",
          source: "",
          web_url: "",
        },
      },
    });

    const { container } = render(
      <Router>
        <Provider store={store}>
          <ArticlePage />
        </Provider>
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
