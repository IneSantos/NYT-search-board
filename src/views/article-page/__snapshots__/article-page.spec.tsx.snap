// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePage atom tests ArticlePage component render 1`] = ` `;
